### PR TITLE
feat: use albedo as backend server

### PR DESCRIPTION
**Pull request by theseion**
Update all tests to produce proper responses

### CODE REVIEW

**Comment by theseion** [comment]
@dune73 @airween Would you mind checking the two failing tests?
- `920100-4`: this test only succeeds because the __backend__ (httpbin) returns 405. Apache is totally fine with the request. This test has comes from Trustwave, so I imagine that the issue has been fixed in Apache a while back. I believe we can simply change the test to expect a 200 and no triggered rule.
- `920272-5`: Similar thing here, although I'm not sure whether the test is meant to be run against a server that is not in proxy mode. Not sure what to do here.

**Comment by airween** [comment]
>   * `920100-4`: this test only succeeds because the **backend** (httpbin) returns 405. Apache is totally fine with the request. This test has comes from Trustwave, so I imagine that the issue has been fixed in Apache a while back.
> 
>   * `920272-5`: Similar thing here, although I'm not sure whether the test is meant to be run against a server that is not in proxy mode. Not sure what to do here.

I was able to reproduce your experiments: with `httpbin` the tests were succeeded, but with `albedo` were not.

Here is the output of `albedo`:
```
HTTP/1.1 200 OK
Date: Thu, 23 May 2024 19:48:25 GMT
Content-Length: 0
```
and `httpbin`:
```
HTTP/1.1 405 METHOD NOT ALLOWED
Server: gunicorn
Date: Thu, 23 May 2024 19:51:31 GMT
Connection: close
Content-Type: text/html; charset=utf-8
Allow: HEAD, OPTIONS, GET
Content-Length: 153
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
```

> I believe we can simply change the test to expect a 200 and no triggered rule.

Does it make sense? This will result in a successful test, but the test will not be valuable. I think we can remove these tests finally.

**Comment by fzipi** [comment]
Just a comment: httpbin has a `/status/NNN` endpoint where it will return the http status code you are sending in the path.

**Comment by dune73** [comment]
Thank you @airween. Nothing to add from my side.

File: tests/docker-compose.yml [hunk 1]
```diff
@@ -68,8 +68,5 @@ services:
       - backend
 
   backend:
-    # Use go-httpbin instead of httpbin to ensure that Unicode bytes in JSON
-    # are not reflected as JSON Unicode escape sequences but as the data
-    # actualy sent.
-    image: mccutchen/go-httpbin
-    command: ["/bin/go-httpbin", "-port", "80"]
+    image: ghcr.io/coreruleset/albedo:0.0.13
```
Comment (M4tteoP) [inline tests/docker-compose.yml:71]: nit: maybe not needed, but you just released `0.0.14`

**Review by M4tteoP — APPROVED** [review]
Looks good to me!

**Review by M4tteoP — APPROVED** [review]

**Pull request merged**